Source/HTTP/Curl/CurlEasyRequest.cpp

```diff
@@ -0,0 +1,345 @@
+
+    auto request = static_cast<CurlEasyRequest*>(context);
+
+#if HC_TRACE_INFORMATION_ENABLE
+    if (size * nitems > 2)
+    {
+        HC_TRACE_INFORMATION(HTTPCLIENT, "'%.*s'", size * nitems - 2, buffer); // -2 to avoid printing \r\n
+    }
+#endif
+
+    size_t bufferSize = size * nitems;
+    char const* current = buffer;
+    char const* end = buffer + bufferSize;
+
+    // scan for the end of the header name
+    char const* name = current;
+    size_t nameSize = 0;
+    for (; current < end; ++current)
+    {
+        if (*current == ':')
+        {
+            nameSize = current - buffer;
+            ++current;
+            break;
+        }
+    }
+    if (current == end)
+    {
+        // not a real header, drop it
+        return bufferSize;
+    }
+
+    // skip whitespace
+    for (; current < end && *current == ' '; ++current) // assume that Curl canonicalizes headers
+    {}
+
+    // scan for the end of the header value
+    char const* value = current;
+    size_t valueSize = 0;
+    char const* valueStart = current;
+    for (; current < end; ++current)
+    {
+        if (*current == '\r')
+        {
+            valueSize = current - valueStart;
+            break;
+        }
+    }
+    if (current == end)
+    {
+        // curl should always gives us the new lines at the end of the header
+        assert(false);
+    }
+
+    HRESULT hr = HCHttpCallResponseSetHeaderWithLength(request->m_hcCallHandle, name, nameSize, value, valueSize);
+    assert(SUCCEEDED(hr));
+
+    return bufferSize;
+}
+
+size_t CurlEasyRequest::WriteDataCallback(char* buffer, size_t /*size*/, size_t nmemb, void* context) noexcept
+{
+    HC_TRACE_VERBOSE(HTTPCLIENT, "CurlEasyRequest::WriteDataCallback: received data (%llu bytes)", nmemb);
+
+    auto request = static_cast<CurlEasyRequest*>(context);
+
+    HC_TRACE_INFORMATION(HTTPCLIENT, "'%.*s'", nmemb, buffer);
+
+    HRESULT hr = HCHttpCallResponseAppendResponseBodyBytes(request->m_hcCallHandle, reinterpret_cast<uint8_t*>(buffer), nmemb);
+    assert(SUCCEEDED(hr));
+
+    return nmemb;
+}
+
+int CurlEasyRequest::DebugCallback(CURL* /*curlHandle*/, curl_infotype type, char* data, size_t size, void* /*context*/) noexcept
+{
+    char const* event = "<unknown>";
+    switch (type)
+    {
+    case CURLINFO_TEXT: event = "TEXT"; break;
+    case CURLINFO_HEADER_OUT: event = "HEADER OUT"; break;
+    case CURLINFO_DATA_OUT: event = "DATA OUT"; break;
+    case CURLINFO_SSL_DATA_OUT: event = "SSL OUT"; break;
+    case CURLINFO_HEADER_IN: event = "HEADER IN"; break;
+    case CURLINFO_DATA_IN: event = "DATA IN"; break;
+    case CURLINFO_SSL_DATA_IN: event = "SSL IN"; break;
+    case CURLINFO_END: event = "END"; break;
+    }
+
+    if (type == CURLINFO_TEXT && data[size - 1] == '\n')
+    {
+        size -= 1;
+    }
+
+    HC_TRACE_IMPORTANT(HTTPCLIENT, "CURL %10s - %.*s", event, size, data);
```

Comment:
This is very spammy and basically includes the entire request + response data.  "Important" might be the wrong classification?

Reply:
You are right this feels more like info

Reply:
changed to HC_TRACE_INFORMATION